content/recipes/authoring/kata-snippets/full-tests.md

```diff
@@ -5,6 +5,84 @@ prev: /recipes/authoring/kata-snippets/sample-tests/
+- If your test suite uses random tests (and it usually should, see below), then make sure that every test scenario generated there has its equivalent fixed test. Fixed tests are usually easier to debug, so ideally a fixed test should fail before the corresponding scenario is tested with random tests.
+
+
+## Random Tests
+
+Random tests are uncommon in "real life" coding and are somewhat specific to Codewars. They are required by many kata to reject solutions based on input probing, hard-coding, and other workarounds. The goal of random tests is to make the inputs unpredictable so that only proper solutions may pass.
+
+- **Each requirement should have at least one (preferably more) corresponding random test case.** Basically, randomly generated scenarios should correspond to the set of fixed tests. You should not rely on luck and hope that random tests will generate invalid inputs and edge conditions by chance.
+- **Random inputs should be _truly_ (pseudo-)random**, i.e. they should be _dynamically_ generated on every run with the help of suitable library functions such as `Math.random()` in JavaScript or `rand()` in C. See this [obligatory XKCD comic](https://xkcd.com/221/) for how NOT to do random tests.
+- In some languages such as C/C++, **make sure that the RNGs are properly seeded with the current time at the beginning** using `srand(time(NULL))` or equivalent; otherwise, the same pseudorandom sequence will be generated on every run and your "random" inputs will devolve into fixed ones.
+- **The order of random tests should be non-deterministic** especially if the expected output is a boolean variable (e.g. checking that some input satisfies some criteria). Not only should inputs be randomized, but their order as well. It should not be the case that, for example, 10 valid, random inputs are tested in the first step followed by 10 invalid ones. The order of tested scenarios should be unpredictable too, and one possible technique is to generate and collect a set of random inputs for all required scenarios and shuffle them.
+- Under some rare circumstances, it's allowed to use so called **randomized tests** instead of fully random ones. For some types of problems (for example, kata related to chess, or problems with small set of possible inputs) it's too complex or infeasible to generate inputs randomly. In such rare cases, it's accepted to use a predefined, hard-coded set of inputs (and possibly expected outputs too), and shuffle them before the tests are run. If possible, some additional transformation can be applied randomly (for example array can be reversed, game board can be rotated, sides can be flipped) if it can be easily accounted for in the result of the test.
```

Comment:
imo, the definitions were a good addition. Here, randomized tests aren't explained.

Mmmmmh, wait... iirc I put something in the glossary already

```suggestion
- Under some rare circumstances, it's allowed to use so called [**randomized tests**](/glossary/#randomized-tests) instead of fully random ones. For some types of problems (for example, kata simulating chess, or problems with small set of possible inputs) it may be too complex or infeasible to generate inputs randomly. In such rare cases, it's accepted to use a predefined, hard-coded set of inputs (and possibly expected outputs too) and shuffle them before the tests are run. If possible, some additional transformation can be applied randomly (for example array can be reversed, game board can be rotated, sides can be flipped) if it can be easily accounted for in the result of the test. Or a small fraction of the hardcoded inputs could be randomly skiped at each run to make the hardcoding of the results more tedious for the user.
```